### PR TITLE
TrackerNano implementation

### PR DESCRIPTION
**Pull request by figamore**
I have added bindings for using OpenCV's TrackerNano single object tracker, which is superior to any of the other available trackers (CSRT, MOSSE, TLD, etc.).

The changes allow a new TrackNano instance to be created by specifying the path of the backbone and neckhead model files in the "new cv.TrackerNano(backboneModelPath: String, neckheadModelPath)" constructor. All other methods, such as init and update, follow the convention and usage of the rest of the trackers.

### CODE REVIEW

File: cc/tracking/Trackers/TrackerNano.cc [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#include "opencv_modules.h"
+
+#ifdef HAVE_OPENCV_TRACKING
+
+#include "TrackerNano.h"
+
+#if CV_VERSION_GREATER_EQUAL(3, 2, 0)
```
Comment (UrielCh) [inline cc/tracking/Trackers/TrackerNano.cc:7]: Think the min value for the NanoTracker is 4.7.0

**Comment by UrielCh** [comment]
- clean cleanup your CV_VERSION_GREATER_EQUAL macro usage.
- add a test unit.

**Comment by figamore** [comment]
> * clean cleanup your CV_VERSION_GREATER_EQUAL macro usage.
> * add a test unit.

Done.

**Comment by UrielCh** [comment]
ho nooooo !!!!

**Comment by figamore** [comment]
> ho nooooo !!!! Some checks were not successful

Oh no! There was an issue with loading the model in the unit test. I have fixed it now.

File: cc/tracking/Trackers/TrackerNano.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#include "../Tracker.h"
+
+#if CV_VERSION_GREATER_EQUAL(3, 2, 0)
```
Comment (UrielCh) [inline cc/tracking/Trackers/TrackerNano.h:3]: 3.2.0 ???

Comment (figamore) [reply cc/tracking/Trackers/TrackerNano.h]: I missed that one 😭 All fixed now!

**Comment by figamore** [comment]
@UrielCh, it looks like the test "Use openCV from Brew / build (4, 22) (pull_request)" failed due to an unrelated system fault. Could you please run it again?

**Comment by UrielCh** [comment]
all Tests pass, but the I still need to test mutiple openCV version.

**Comment by UrielCh** [comment]
I still get some Stack trace whene I use TrackerNano.

**Comment by figamore** [comment]
> I still get some Stack trace whene I use TrackerNano.

Did you try it using the official TrackerNano model files here?

https://github.com/HonglinChu/SiamTrackers/tree/master/NanoTrack/models/nanotrackv2

**Comment by UrielCh** [comment]
I did some cleaning in the source tree moving all legacy tracker in a single file.

**Comment by UrielCh** [comment]
you commit the 2 models:

fdf592d3f946aa84775eecd1e388ab9f  nanotrackv2/nanotrack_backbone_sim.onnx
7eb4bfeda369ece80218ef4070b9bef6  nanotrackv2/nanotrack_head_sim.onnx

is there something wrong with the V3 ?

could you please add an smple code in the "examples" folder ?

by the way the `ackbone.onnx` and `neckhead.onnx` file should be moved to the `data` folder

**Comment by figamore** [comment]
> you commit the 2 models:
> 
> 
> 
> fdf592d3f946aa84775eecd1e388ab9f  nanotrackv2/nanotrack_backbone_sim.onnx
> 
> 7eb4bfeda369ece80218ef4070b9bef6  nanotrackv2/nanotrack_head_sim.onnx
> 
> 
> 
> is there something wrong with the V3 ?
> 
> 

I haven't tried the V3 models, but they should work just as well. I only included the V2 models so the tests run could run.

**Comment by UrielCh** [comment]
could you please add an smple code in the "examples" folder ?

by the way the `ackbone.onnx` and `neckhead.onnx` file should be moved to the `data` folder

Use the branch named 7.2.0 to make changes

https://github.com/UrielCh/opencv4nodejs/tree/7.2.0

**Pull request merged**